File: MoveZeroes/move_zeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        int i = 0;
        int j = 0;
        int n = nums.Length;
        while (j < n) {
            nums[i] = nums[j];
            j++;
            if (nums[i] != 0)
                i++;
        }
        while (i < n) {
            nums[i] = 0;
            i++;
        }
    }
}

File: MoveZeroes/move_zeroes.py
class Solution(object):
    
    def moveZeroes(self, nums):
        i, j, n = 0, 0, len(nums)
        while j < n:
            nums[i] = nums[j]
            j += 1
            if nums[i] != 0:
                i += 1
        while i < n:
            nums[i] = 0
            i += 1